--- a/luminol-server/build.gradle.kts
+++ b/luminol-server/build.gradle.kts
@@ -50,7 +_,23 @@
         }
     }
 
-    activeFork = luminol
+    val lophine = forks.register("lophine") {
+        forks = luminol
+        upstream.patchRepo("paperServer") {
+            upstreamRepo = luminol.patchedRepo("paperServer")
+            patchesDir = rootDirectory.dir("lophine-server/paper-patches")
+            outputDir = rootDirectory.dir("paper-server")
+        }
+
+        upstream.patchDir("luminolServer") {
+            upstreamPath = "luminol-server"
+            excludes = setOf("src/minecraft", "paper-patches", "minecraft-patches", "build.gradle.kts", "build.gradle.kts.patch")
+            patchesDir = rootDirectory.dir("lophine-server/luminol-patches")
+            outputDir = rootDirectory.dir("luminol-server")
+        }
+    }
+
+    activeFork = lophine
 
 
 
@@ -125,12 +_,16 @@
         resources { srcDir("../paper-server/src/main/resources") }
         java { srcDir("../folia-server/src/main/java") }
         resources { srcDir("../folia-server/src/main/resources") }
+        java { srcDir("../luminol-server/src/main/java") }
+        resources { srcDir("../luminol-server/src/main/resources") }
     }
     test {
         java { srcDir("../paper-server/src/test/java") }
         resources { srcDir("../paper-server/src/test/resources") }
         java { srcDir("../folia-server/src/test/java") }
         resources { srcDir("../folia-server/src/test/resources") }
+        java { srcDir("../luminol-server/src/main/java") }
+        resources { srcDir("../luminol-server/src/main/resources") }
     }
 }
 val log4jPlugins = sourceSets.create("log4jPlugins") {
@@ -158,10 +_,10 @@
 }
 
 dependencies {
-    implementation(project(":luminol-api"))
+    implementation(project(":lophine-api"))
     // Luminol start - Dependenices insert
     implementation("net.objecthunter:exp4j:0.4.8")
-    implementation("com.electronwill.night-config:toml:3.9.0") // Night config
+    implementation("com.electronwill.night-config:toml:3.8.4") // Night config
     implementation("com.github.luben:zstd-jni:1.5.4-1")
     implementation("net.openhft:zero-allocation-hashing:0.16")
     implementation("io.github.classgraph:classgraph:4.8.158")
@@ -234,14 +_,14 @@
         val gitBranch = git.exec(providers, "rev-parse", "--abbrev-ref", "HEAD").get().trim()
         attributes(
             "Main-Class" to "org.bukkit.craftbukkit.Main",
-            "Implementation-Title" to "Luminol", // Luminol - Rebrand
+            "Implementation-Title" to "Lophine", // Luminol - Rebrand
             "Implementation-Version" to implementationVersion,
             "Implementation-Vendor" to date,
-            "Specification-Title" to "Luminol", // Luminol - Rebrand
+            "Specification-Title" to "Lophine", // Luminol - Rebrand
             "Specification-Version" to project.version,
             "Specification-Vendor" to "LuminoLMC", // Luminol - Rebrand
-            "Brand-Id" to "luminolmc:luminol", // Luminol - Rebrand
-            "Brand-Name" to "Luminol", // Luminol - Rebrand
+            "Brand-Id" to "luminolmc:lophine", // Luminol - Rebrand
+            "Brand-Name" to "Lophine", // Luminol - Rebrand
             "Build-Number" to (build ?: ""),
             "Build-Time" to buildTime.toString(),
             "Git-Branch" to gitBranch,
